standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -10330,6 +10330,108 @@ public boolean deletePartitionColumnStatistics(String catName, String dbName, St
     return ret;
   }
 
+  @Override
+  public boolean deletePartitionMultiColumnStatistics(String catName, String dbName, String tableName,
+                                                 String partName, List<String> partVals, List<String> colNames, String engine)
+          throws NoSuchObjectException, MetaException, InvalidObjectException, InvalidInputException {
+    boolean ret = false;
+    Query query = null;
+    dbName = org.apache.commons.lang3.StringUtils.defaultString(dbName,
+            Warehouse.DEFAULT_DATABASE_NAME);
+    catName = normalizeIdentifier(catName);
+    if (tableName == null) {
+      throw new InvalidInputException("Table name is null.");
+    }
+    try {
+      openTransaction();
+      MTable mTable = getMTable(catName, dbName, tableName);
+      MPartitionColumnStatistics mStatsObj;
+      List<MPartitionColumnStatistics> mStatsObjColl;
+      if (mTable == null) {
+        throw new NoSuchObjectException("Table " + tableName
+                + "  for which stats deletion is requested doesn't exist");
+      }
+      // Note: this does not verify ACID state; called internally when removing cols/etc.
+      //       Also called via an unused metastore API that checks for ACID tables.
+      MPartition mPartition = getMPartition(catName, dbName, tableName, partVals, mTable);
+      if (mPartition == null) {
+        throw new NoSuchObjectException("Partition " + partName
+                + " for which stats deletion is requested doesn't exist");
+      }
+      query = pm.newQuery(MPartitionColumnStatistics.class);
+      String filter;
+      String parameters;
+      if (colNames != null) {
+        filter = "partition.partitionName == t1 && partition.table.database.name == t2 && partition.table.tableName == t3 && "
+                + "t4.contains(colName) && partition.table.database.catalogName == t5" + (engine != null ? " && engine == t6" : "");
+        parameters = "java.lang.String t1, java.lang.String t2, java.lang.String t3, "
+                + "java.util.Collection t4, java.lang.String t5" + (engine != null ? ", java.lang.String t6" : "");
+      } else {
+        filter = "partition.partitionName == t1 && partition.table.database.name == t2 && partition.table.tableName == t3 && partition.table.database.catalogName == t4" + (engine != null ? " && engine == t5" : "");
+        parameters = "java.lang.String t1, java.lang.String t2, java.lang.String t3, java.lang.String t4" + (engine != null ? ", java.lang.String t5" : "");
+      }
+      query.setFilter(filter);
+      query.declareParameters(parameters);
+      if (colNames != null) {
+        query.setUnique(true);
+        for (String colName : colNames){
+          // trim the extra spaces, and change to lowercase
+          normalizeIdentifier(colName);
```

Comment:
need to put the normalized column names into new list which will be passed to `executeWithArray`

Reply:
Acknowledged. Created a new list `normalizedColNames`, to pass to `executeWithArray`.